core/network/impl/synchronizer_impl.hpp

```diff
@@ -255,6 +207,16 @@ namespace kagome::network {
     void setHangTimer();
     void onHangTimer();
 
+    bool addHeader(const PeerId &peer_id, primitives::BlockData &&data);
+    void fetchTasks();
+    void visitAncestry(const BlockInfo &root, const auto &f);
+    bool canFetchMore(size_t &fetching) const;
+    bool validate(const BlocksRequest &request, BlocksResponse &response);
```

Comment:
Can be const?
Also, not clear why response cannot be const reference

Reply:
Ok, will try to add `const`.

`response` is mutable to compute and save header `hash` field.

Reply:
Anyway we can make only `hash` field is mutable